progressbar.go

```diff
@@ -159,13 +173,41 @@ func (p *ProgressBar) Add(num int) error {
 	}
 
 	if updateBar {
-		return renderProgressBar(p.config, p.state)
+		return p.render()
+	}
+
+	return nil
+}
+
+// Clear erases the progress bar from the current line
+func (p *ProgressBar) Clear() error {
+	return clearProgressBar(p.config, p.state)
+}
+
+// render renders the progress bar, updating the maximum
+// rendered line width. this function is not thread-safe,
+// so it must be called with an acquired lock.
+func (p *ProgressBar) render() error {
+	// first, clear the existing progress bar
+	err := clearProgressBar(p.config, p.state)
```

Comment:
@CrushedPixel I just noticed that this line is not working in Windows. It causes the bar to start pushing lots of new lines after it hits ~50%. I don't know why. It does work fine on Linux though. Do you know why this would happen?

Otherwise everything looks awesome. If you don't know about this Windows snafu also, no problem, I will merge anyway and add a wrapper to check if Windows to support that line.

Reply:
What's your setup of the ProgressBar so I can try and replicate the issue myself?

Reply:
Oh interesting - it depends on the size of the window, even though the window it still way bigger than the bar, it will add new lines. Here's a screencast: https://screencast-o-matic.com/watch/cF1QotFVWJ

I'm pretty sure that new lines should only be added when the window is smaller than the bar.

Reply:
My assumption is that the character you're using for the Saucer is more than a single rune, therefore increasing the string length twofold. What character exactly are you using?

Reply:
I can reproduce the issue in my Mac Terminal when using the unicode block █ as the saucer. Will try and fix.

Reply:
The issue is exactly what I assumed - the unicode block █ counts for 3 individual characters...
https://play.golang.org/p/0czfwHE073F

Reply:
Oooh great catch! Thanks. So basically the "count" of characters just needs to be changed

Reply:
Casting the string to a slice of runes before getting the length seems to do the trick:
https://play.golang.org/p/RcIuaflf0oY